libs/cdk/virtual-table/src/services/selection/selection.service.ts

```diff
@@ -86,20 +87,25 @@ export class SelectionService<T> implements OnDestroy {
         this.checkIsAllSelected();
     }
 
+    // eslint-disable-next-line complexity,max-lines-per-function
     public selectRow(row: Nullable<T>, event: MouseEvent): void {
         if (isNil(row)) {
             return;
         }
 
+        if (isNil(this.userAgentService.userOS)) {
+            this.userAgentService.detectOS();
+        }
+
         const rows: T[] = this.rows ?? [];
-        const { shiftKey, ctrlKey }: MouseEvent = event;
+        const { shiftKey, ctrlKey, metaKey }: MouseEvent = event;
         const index: number = rows.findIndex(
             (item: T): boolean => ((item as any) ?? ({} as T))[this.primaryKey] === (row ?? {})[this.primaryKey]
         );
 
         if (shiftKey) {
             this.multipleSelectByShiftKeydown(index);
-        } else if (ctrlKey) {
+        } else if ((this.userAgentService.isMacOS() && metaKey) || (!this.userAgentService.isMacOS() && ctrlKey)) {
```

Comment:
это можно написать проще:
```ts
if (this.userAgentService.isMacOS() ? metaKey : ctrlKey) 
```

Reply:
why you use service instead of minimal function for that?

```ts
import { isMacOS } from `@angular-ru/common/platform`; // or `@angular-ru/common/utils`

// ...

if (isMacOS() ? metaKey : ctrlKey) {
 // ..
}
```

```ts
export function isMacOS(navigatorRef: Navigator = navigator): boolean { 
  return navigatorRef.userAgent.includes('Mac');
}
```

platform.spec.ts

```ts
expect(isMacOS({ userAgent: `Mac` } as unknown as Navigator).toBeTruthy();
expect(isMacOS({ userAgent: `Linux` } as unknown as Navigator).toBeFalsy();
```

Reply:
in general, a service was planned for use in the future to calculate not only mac os, but it turned out somehow neither here nor there. @malvill, I propose to do it as simply as shown by Max